File: Tree/129. Sum Root to Leaf Numbers/tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum(TreeNode* root, int num)
    {
        if(root == NULL)
        {
            return 0;
        }
        num = num*10;
        num += root->val;
        if(root->left == NULL && root->right ==NULL)
        {
            return num;
        }
        int left = sum(root->left,num);
        int right = sum(root->right,num);
        return left+right;
    }
    int sumNumbers(TreeNode* root) 
    {
        int num;
        return sum(root,  num=0);
    }
};

File: Tree/129. Sum Root to Leaf Numbers/tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        
        def preorder(root, num):
            if root is None:
                return 0
            num = num*10
            num += root.val
            if not root.left and not root.right: #leaf node
                return num
            left = preorder(root.left, num) #495 #495+491
            right = preorder(root.right, num) #491 #40
            return left+right # 495+491+40

        return preorder(root, 0)